design/TextSearch.md

```diff
@@ -0,0 +1,49 @@
+# Text Search Design
+
+(aka "BigSearch")
+
+## Key ideas
+
+- Users create text files that provide text data to index and search.
+- Search indexing builds internal resources for faster query results.
+- Each search query leads to zero or more search matches.
+- Each search match includes an internal file position.
```

Comment:
What is a file position? Byte offset, Motoko Character offset?

Reply:
Good question,  I was thinking Motoko `Char`s, starting at `0` in each file.

But stepping back, I think the positions can be more general "coordinates" and can even have redundant dimensions (file id, position in file by char offset, position in file by line/char pair).

To start, I'd focus on Motoko counting the `Char`s in each "file", and using those as the positions.  I realize that's not best practices for Unicode; is there an alternative?